---
title: "Capstone 2 analysis"
author: "Eugenio Mattei UMASS medical School"
date: "Run on `r Sys.Date()`"
documentclass: article
output: 
  fontsize: 14pt
  html_document:
  toc: true
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(eval=TRUE, echo=FALSE, error=FALSE, warning = FALSE) 

rm(list=ls())

library(data.table)
library(ggplot2)
library(dplyr)
library(synapseClient)
library(edgeR)

synapseLogin()

#counts syn11305250
#metadata syn11305353

tol21rainbow= c("#771155", "#AA4488", "#CC99BB", "#114477", "#4477AA", "#77AADD", "#117777", "#44AAAA", "#77CCCC", "#117744", "#44AA77", "#88CCAA", "#777711", "#AAAA44", "#DDDD77", "#774411", "#AA7744", "#DDAA77", "#771122", "#AA4455", "#DD7788")
rich12equal = c("#000040", "#000093", "#0020E9", "#0076FF", "#00B8C2", "#04E466", "#49FB25", "#E7FD09", "#FEEA02", "#FFC200", "#FF8500", "#FF3300")
mycols=c("dodgerblue2", "#E31A1C", "green4", "#6A3D9A", "#FF7F00", "black",
         "gold1", "skyblue2", "#FB9A99", "palegreen2", "#CAB2D6", "#FDBF6F",
         "gray70", "khaki2", "maroon", "orchid1", "deeppink1", "blue1",
         "steelblue4", "darkturquoise", "green1", "yellow4", "yellow3",
         "darkorange4", "brown")
```

Loading the Capstone2 metadata
```{r}
capstone2.metadata = fread(getFileLocation(synGet( 'syn11305353' )), header=TRUE, stringsAsFactors=FALSE,sep="\t",data.table = F) 
capstone2.metadata$has_GENOTYPE_bool=TRUE
capstone2.metadata$has_GENOTYPE_bool[capstone2.metadata$has_GENOTYPE=="FALSE"]=FALSE
head(capstone2.metadata)
```

*ChIPseq Statistics*
```{r}
temp = capstone2.metadata %>% group_by(Brain_Region,Cell_Type,Mark) %>% summarise(n=n())
temp
ggplot(temp,aes(as.factor(Cell_Type),n,fill=Mark))+
  geom_col(position='dodge')+
  theme_bw()+
  scale_fill_manual(values = mycols)+
  ylab("Number of samples")+
  xlab("")+
  ggtitle("Peaks Summary")+
  geom_label(data=temp, aes(y=n,colour=Mark,label = n), size = 5,position = position_dodge(width = 0.90),vjust=-0.5,color="white")+
  facet_wrap(~Brain_Region)

temp = capstone2.metadata %>% group_by(Cell_Type,Mark) %>% summarise(n=n())
ggplot(temp,aes(as.factor(Cell_Type),n,fill=Mark))+
  geom_col(position='dodge')+
  theme_bw()+
  scale_fill_manual(values = mycols)+
  ylab("Number of samples")+
  xlab("")+
  ggtitle("Peaks Per Cell Type")+
  geom_label(data=temp, aes(y=n,colour=Mark,label = n), size = 5,position = position_dodge(width = 0.90),vjust=-0.5,color="white")

temp = capstone2.metadata %>% group_by(Brain_Region,Mark) %>% summarise(n=n())
ggplot(temp,aes(as.factor(Brain_Region),n,fill=Mark))+
  geom_col(position='dodge')+
  theme_bw()+
  scale_fill_manual(values = mycols)+
  ylab("Number of samples")+
  xlab("")+
  ggtitle("Peaks Per Brain Region")+
  geom_label(data=temp, aes(y=n,colour=Mark,label = n), size = 5,position = position_dodge(width = 0.90),vjust=-0.5,color="white")


temp = capstone2.metadata %>% group_by(Study,Brain_Region,Mark) %>% summarise(n=n())
ggplot(temp,aes(as.factor(Study),n,fill=Mark))+
  geom_col(position='dodge')+
  theme_bw()+
  scale_fill_manual(values = mycols)+
  ylab("Number of samples")+
  xlab("")+
  ggtitle("Number of Samples Per Study")+
  geom_label(data=temp, aes(y=n,colour=Mark,label = n), size = 5,position = position_dodge(width = 0.90),vjust=-0.5,color="white")+
  facet_wrap(~Brain_Region)

```

*ATACseq Summary*
```{r}
temp = capstone2.metadata %>% group_by(Brain_Region,has_ATACseq) %>% summarise(n=n())
ggplot(temp,aes(as.factor(Brain_Region),n,fill=has_ATACseq))+
  geom_col(position='dodge')+
  theme_bw()+
  scale_fill_manual(values = c("darkred","darkgreen"),name="Has ATACseq")+
  ylab("Number of samples")+
  xlab("")+
  ggtitle("ATACseq Statistics")+
  geom_label(data=temp, aes(y=n,colour=has_ATACseq,label = n), size = 5,position = position_dodge(width = 0.90),vjust=-0.5,color="white")
```

*RNAseq summary*
```{r}
temp = capstone2.metadata %>% group_by(Brain_Region,has_RNAseq) %>% summarise(n=n())
ggplot(temp,aes(as.factor(Brain_Region),n,fill=has_RNAseq))+
  geom_col(position='dodge')+
  theme_bw()+
  scale_fill_manual(values = c("darkred","darkgreen"),name="Has RNAseq")+
  ylab("Number of samples")+
  xlab("")+
  ggtitle("RNAseq Statistics")+
  geom_label(data=temp, aes(y=n,colour=has_RNAseq,label = n), size = 5,position = position_dodge(width = 0.90),vjust=-0.5,color="white")

  
```

*Genotype summary*
```{r}
temp = capstone2.metadata %>% group_by(Brain_Region,has_GENOTYPE_bool) %>% summarise(n=n())
ggplot(temp,aes(as.factor(Brain_Region),n,fill=has_GENOTYPE_bool))+
  geom_col(position='dodge')+
  theme_bw()+
  scale_fill_manual(values = c("darkred","darkgreen"),name="Has Genotype")+
  ylab("Number of samples")+
  xlab("")+
  ggtitle("Genotype Statistics")+
  geom_label(data=temp, aes(y=n,colour=has_GENOTYPE_bool,label = n), size = 5,position = position_dodge(width = 0.90),vjust=-0.5,color="white")

```



#Counts Stats
```{r}
capstone2.h3k4me3.counts = fread(getFileLocation(synGet( 'syn11305250' )), header=TRUE, stringsAsFactors=FALSE,sep="\t",data.table = F) 

rownames(capstone2.h3k4me3.counts)=capstone2.h3k4me3.counts$Geneid
capstone2.h3k4me3.counts=capstone2.h3k4me3.counts[,-1]
head(capstone2.h3k4me3.counts)

#4 samples missing
counts.reduced = capstone2.h3k4me3.counts[,names(capstone2.h3k4me3.counts) %in% capstone2.metadata$Sample_ID]
capstone2.metadata.sorted=capstone2.metadata[match(names(counts.reduced),capstone2.metadata$Sample_ID),]
head(capstone2.metadata.sorted)


#cpm = cpm(capstone2.h3k4me3.counts,log = F)
#cpm.log = cpm(capstone2.h3k4me3.counts,log=T)
cpm = cpm(counts.reduced,log = F)
cpm.log = cpm(counts.reduced,log=T)

```

##Clustering to test for separation between Cell Types

```{r,message=F,fig.width=9,fig.height=12}
library(devtools)
library(RColorBrewer)
library(rafalib)

#Load latest version of heatmap.3 function
source_url("https://raw.githubusercontent.com/obigriffith/biostar-tutorials/master/Heatmaps/heatmap.3.R")
hmcol <- colorRampPalette(brewer.pal(9, "GnBu"))(100)


cols <- palette(brewer.pal(8, "Dark2"))[as.fumeric(as.character(capstone2.metadata.sorted$Brain_Region))]
cols2 <- mycols[as.fumeric(as.character(capstone2.metadata.sorted$Cell_Type))]


colLab=t(cbind(cols,cols2))
rownames(colLab)=c("brainRegion","cellType")
#colnames(colLab)=names(capstone2.h3k4me3.counts)
colnames(colLab)=names(counts.reduced)

test_cormat <- round(cor(cpm.log,method = "spearman"),2)
test_dist = as.dist((1-test_cormat)/2)
heatmap.3(test_dist,labCol=FALSE,scale="none",
          trace="none", 
          ColSideColors = t(colLab), KeyValueName="Spearman Correlation",
          col=hmcol,ColSideColorsSize=3)

legend("topleft",
       legend=c("NeuN+","NeuN-","Homogenate","ACC","DLPFC","CNON"), 
       fill=c(unique(cols2),unique(cols)), ncol=2)

```

PCA

```{r}
counts.pca <- prcomp(t(cpm), center = TRUE, scale. = TRUE)

#palette(brewer.pal(8, "Dark2"))
cols <- mycols[as.fumeric(as.character(capstone2.metadata.sorted$Brain_Region))]
cols2 <- mycols[as.fumeric(as.character(capstone2.metadata.sorted$Cell_Type))]
cols3 <- mycols[as.fumeric(as.character(capstone2.metadata.sorted$Study))]

plot(counts.pca$x[,1:2], col = cols)
legend("topright", inset=.05, title="Cell Types",c("ACC","DLPFC","CNON"), fill=unique(cols), horiz=F)


plot(counts.pca$x[,1:2], col = cols2)
legend("topright", inset=.05, title="Cell Types",c("NeuN+","NeuN-","Homogenate"), fill=unique(cols2), horiz=F)

plot(counts.pca$x[,1:2], col = cols3)
legend("topright", inset=.05, title="Cell Types",unique(capstone2.metadata.sorted$Study), fill=unique(cols2), horiz=F)

```



